### PR TITLE
Add show/hide gc button in memory screen(#1089)

### PR DESCRIPTION
**Pull request by alexlindroos**
PR #1089

### CODE REVIEW

File: packages/devtools_app/lib/src/memory/memory_screen.dart [hunk 1]
```diff
@@ -1526,6 +1541,25 @@ class MemoryConfigurationsDialog extends StatelessWidget {
...
               ],
             ),
+            const SizedBox(height: defaultSpacing,),
+            ...dialogSubHeader(theme, 'General'),
+            Column(
+              children: [
+                Row(
+                  children: [
+                    NotifierCheckbox(
+                        notifier: controller.advancedSettingsEnabled),
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/memory/memory_screen.dart:1551]: nit comma after single parameter e.g., notifier: controller.advancedSettingsEnabled,)

**Comment by terrylucas** [comment]
Thanks Alex.  A few tests will need to be change.  In particular, test/memory_screen_test.dart:

Expects to find the GC button.  Probably good idea to expect it by default to not be there and also if the enabledAdvancedMemory is set then:

      expect(find.byKey(MemoryScreen.gcButtonKey), findsOneWidget);

Also, the golden files (screenshots of the current visual state of DevTools at a particular point in a test) need to be updated in the default case of advanced is not enabled.

Probably good to add a golden file test when advanced is enabled too.

To update the golden masters run the test with the flag --update-goldens e.g.,

flutter test --update-goldens test/memory_screen_test.dart

will update all golden files to compare then the tests should pass.  If you'd like me to fix up the tests and goldens let me know?

**Comment by alexlindroos** [comment]
Hi @terrylucas, thanks for the review. I updated the memory_screen_test, is it okay now? I am not sure about how to add the golden file test, so if you want you can add that. Thank you!

File: packages/devtools_app/lib/src/memory/memory_screen.dart (hunk 1 above)
Comment (alexlindroos) [reply packages/devtools_app/lib/src/memory/memory_screen.dart]: fixed

File: packages/devtools_app/lib/src/memory/memory_screen.dart [hunk 2]
```diff
@@ -1526,6 +1541,25 @@ class MemoryConfigurationsDialog extends StatelessWidget {
...
               ],
             ),
+            const SizedBox(height: defaultSpacing,),
+            ...dialogSubHeader(theme, 'General'),
+            Column(
+              children: [
+                Row(
+                  children: [
+                    NotifierCheckbox(
+                        notifier: controller.advancedSettingsEnabled,),
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/memory/memory_screen.dart:1551]: dartformat needs to be run. Then the line will be:

```
                    NotifierCheckbox(
                      notifier: controller.advancedSettingsEnabled,
                    ),
```

Comment (alexlindroos) [reply packages/devtools_app/lib/src/memory/memory_screen.dart]: Sorry, I forgot to do that

**Comment by terrylucas** [comment]
I noticed you updated the golden PNG files however, the goldens aren't matching.  I patched in you PR locally and noticed the same.  After I ran with --update-goldens the PNGs changed and now the test will pass (once I add the updated PNGs).

What platform are you running on Flutter, OS, etc?

**Comment by alexlindroos** [comment]
> I noticed you updated the golden PNG files however, the goldens aren't matching. I patched in you PR locally and noticed the same. After I ran with --update-goldens the PNGs changed and now the test will pass (once I add the updated PNGs).
> 
> What platform are you running on Flutter, OS, etc?
> 
> My [PR](https://github.com/flutter/devtools/pull/2789) (your PR with my updated PNG files).

flutter doctor

```
[✓] Flutter (Channel stable, 2.0.0, on macOS 11.1 20C69 darwin-x64, locale en-FI)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
[✓] Connected device (3 available)

• No issues found!
```

**Review by terrylucas — APPROVED** [review]
Submit your PR I'll follow up with the updated goldens.

**Comment by alexlindroos** [comment]
> Submit your PR I'll follow up with the updated goldens.

Sorry to ask, but what do you mean about submitting my PR?

**Pull request merged**